### PR TITLE
WIP: U2F implementaion

### PR DESCRIPTION
**Pull request by MatthewWilkes**

### CODE REVIEW

**Comment by MatthewWilkes** [comment]
Updated against main, skeleton of U2FHID transport in place, with options to enable/disable Keyboard+Mouse and U2FHID independently. Both together not currently working. Would like to merge to get the base in, and avoid bitrot.

**Comment by tomsci** [comment]
I'm not an expert on USB so can't vouch for its correctness but the code looks fine other than minor comments, assuming it doesn't break any existing USB functionality like REPL, no objections from me to merging it.

**Review by tomsci — APPROVED** [review]

File: drivers/tidal_usb/tidal_usb_u2f.c [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        .init.BCNTL = 17,
+    };
+    // Zero out the data packet and then copy in the (17 byte)
+    // response and set the length flags.
+    memset(response.init.data, 0, sizeof response.init.data);
+    memcpy(response.init.data, &response_data, 17);
+    
+    // Cast this to a char array so we can debug print it
+    uint8_t *as_buf = (uint8_t *) &response;
+    for (int i=0;i<64;i++) {
```
[the remark this answers is not in the document]
Comment (MatthewWilkes) [reply drivers/tidal_usb/tidal_usb_u2f.c:126]: There's a macro to use instead, forgot to change it back

File: drivers/tidal_usb/tidal_usb_u2f.c [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    // Allocate a response report
+    u2f_hid_msg response = {
+        .CID = CID_BROADCAST,
+        .init.CMD = U2FHID_INIT,
+        .init.BCNTH = 0,
+        .init.BCNTL = 17,
+    };
+    // Zero out the data packet and then copy in the (17 byte)
+    // response and set the length flags.
+    memset(response.init.data, 0, sizeof response.init.data);
```
[the remark this answers is not in the document]
Comment (MatthewWilkes) [reply drivers/tidal_usb/tidal_usb_u2f.c:121]: Good point

File: tildamk6/sdkconfig.board [hunk 3]
```diff
@@ -21,5 +21,13 @@ CONFIG_TINYUSB_HID_BUFSIZE=128
 CONFIG_TINYUSB_DESC_HID_STRING="TiDAL badge"
 CONFIG_TINYUSB_DESC_MANUFACTURER_STRING="Electromagnetic Field"
 CONFIG_TINYUSB_DESC_PRODUCT_STRING="TiDAL"
+
+CONFIG_TINYUSB_DESC_CUSTOM_VID=0x16D0
+CONFIG_TINYUSB_DESC_CUSTOM_PID=0x1114
+CONFIG_TINYUSB_DESC_USE_ESPRESSIF_VID=
+CONFIG_TINYUSB_DESC_USE_DEFAULT_PID=
+
+CONFIG_LOG_DEFAULT_LEVEL_INFO=y
```
[the remark this answers is not in the document]
Comment (MatthewWilkes) [reply tildamk6/sdkconfig.board:30]: I found it really helpful here, but I'm easy about merging that or not

**Review by tomsci — APPROVED** [review]

**Pull request merged**